### DISCUSSION TITLE
webgpureport.org crashes servo

### DISCUSSION THREAD

**Discussion by beaufortfrancois** [servo/servo#30914] [Q&A] [answered]
When running Servo 0.0.1-5e8f70f with `--pref dom.webgpu.enabled`, it fails to render https://webgpureport.org/

![image](https://github.com/servo/servo/assets/634478/bcdfd461-5a7b-47bb-92b5-14ce6b5f496c)

Without the pref, it loads and simply says WebGPU is not supported.

![image](https://github.com/servo/servo/assets/634478/0c564e57-905c-4d58-9055-fa5d7467319b)

Is this a known issue?

**Comment by jdm** [comment]
Is there any output in your terminal window?

**Reply by beaufortfrancois** [reply]
Here it is:
```
$ /Applications/Servo.app/Contents/MacOS/servo --pref dom.webgpu.enabled https://webgpureport.org
2023-12-22 15:53:14.784 servo[13975:3432175] WARNING: Secure coding is not enabled for restorable state! Enable secure coding by implementing NSApplicationDelegate.applicationSupportsSecureRestorableState: and returning YES.
UNSUPPORTED (log once): POSSIBLE ISSUE: unit 1 GLD_TEXTURE_INDEX_2D is unloadable and bound to sampler type (Float) - using zero texture because texture unloadable
called `Result::unwrap()` on an `Err` value: RecvError (thread Script(1,1), at components/script/script_thread.rs:1498)
[2023-12-22T14:53:15Z ERROR servo::main2] called `Result::unwrap()` on an `Err` value: RecvError
```

**Reply by beaufortfrancois** [reply]
FYI https://demo.servo.org/experiments/webgpu-game-of-life/ works

![image](https://github.com/servo/servo/assets/634478/abc17b6b-fae5-4753-af1f-5957a075db73)

```
$ /Applications/Servo.app/Contents/MacOS/servo --pref dom.webgpu.enabled https://demo.servo.org/experiments/webgpu-game-of-life/
2023-12-22 15:57:59.875 servo[14319:3436272] WARNING: Secure coding is not enabled for restorable state! Enable secure coding by implementing NSApplicationDelegate.applicationSupportsSecureRestorableState: and returning YES.
UNSUPPORTED (log once): POSSIBLE ISSUE: unit 1 GLD_TEXTURE_INDEX_2D is unloadable and bound to sampler type (Float) - using zero texture because texture unloadable
```

**Comment by beaufortfrancois** [comment]
I played a bit more and succeeded to make servo **not** crash when removing the WebGPU API javascript call below:
```js
await navigator.gpu.requestAdapter({ forceFallbackAdapter: true })
```
![image](https://github.com/servo/servo/assets/634478/79e9a939-09cc-470d-a6ea-125a44ab4631)

**Reply by beaufortfrancois** [reply]
According to https://github.com/servo/servo/commit/71e0372ac18709da66e581cf4a777a7cc5d4cb92, @sagudev may be interested in this crash.

**Comment by sagudev** [comment, marked as answer]
I opened https://github.com/servo/servo/issues/30915, but something like this is expected because our webgpu support is still very experimental.

Thanks for detailed report.
